### PR TITLE
NMS-16028: Allow metadata in snmp-config.xml

### PR DESCRIPTION
**Pull request by christianpape**
JIRA: https://opennms.atlassian.net/browse/NMS-16028

### CODE REVIEW

File: core/snmp/api/src/main/java/org/opennms/netmgt/snmp/SnmpProfileMapper.java [hunk 1]
```diff
@@ -42,7 +42,11 @@ public interface SnmpProfileMapper {
      * @param oid         OID with which SNMP get needs to be performed.
      * @return snmpAgentConfig, a @{@link SnmpAgentConfig} from the matching profile.
      */
-    CompletableFuture<Optional<SnmpAgentConfig>> getAgentConfigFromProfiles(InetAddress inetAddress, String location, String oid);
+    default CompletableFuture<Optional<SnmpAgentConfig>> getAgentConfigFromProfiles(InetAddress inetAddress, String location, String oid) {
+        return getAgentConfigFromProfiles(inetAddress, location, oid, true);
+    }
```
Comment (RangerRick) [inline core/snmp/api/src/main/java/org/opennms/netmgt/snmp/SnmpProfileMapper.java:47]: such a simple addition to Java but god this makes interfaces so much nicer

**Review by RangerRick — APPROVED** [review]
looks good

**Comment by christianpape** [comment]
1. I tested with and without encryption enabled. So, if used, the metadata expression will be stored encrypted and interpolated after decryption.
2. What do you mean by "merge" exactly? The same metadata expressions point to the same community in SCV. Of course if you specify definition with the real community and one with the reference to the SCV you cannot merge these.
3. The feature just uses the global SCV scope, so no data for nodes, interfaces or services are available. 
4. yes, sure

What do you think?

**Comment by cgorantla** [comment]
(1) Good that mixing works but I feel these are two different options for user.
For (2),  we merge two definitions if they are same( by merging two successive IP addresses as range). When I use metadata in snmp-config.xml , Does this still work ?
(3) Should be added to docs.

**Comment by christianpape** [comment]
> (1) Good that mixing works but I feel these are two different options for user.
> For (2), we merge two definitions if they are same( by merging two successive IP addresses as range). When I use metadata in snmp-config.xml , Does this still work ?
> (3) Should be added to docs.

See https://github.com/OpenNMS/opennms/pull/6662 for these changes.

**Pull request merged**